src/test/java/com/lambdaschool/foundation/controllers/UserControllerTest.java

```diff
@@ -0,0 +1,268 @@
+            .thenReturn(userList.get(1));
+
+        RequestBuilder rb = MockMvcRequestBuilders.get(apiUrl)
+            .accept(MediaType.APPLICATION_JSON);
+        MvcResult r = mockMvc.perform(rb)
+            .andReturn(); // this could throw an exception
+        String tr = r.getResponse()
+            .getContentAsString();
+
+        ObjectMapper mapper = new ObjectMapper();
+        String er = mapper.writeValueAsString(userList.get(1));
+
+        assertEquals(er,
+            tr,
+            "Expected: " + er + "\nActual: " + tr);
+    }
+
+    @Test
+    void getUserByName() throws
+                         Exception {
+        String apiUrl = "/users/user/name/Arthur";
+
+        Mockito.when(userService.findByName("Arthur"))
+            .thenReturn(userList.get(1));
+
+        RequestBuilder rb = MockMvcRequestBuilders.get(apiUrl)
+            .accept(MediaType.APPLICATION_JSON);
+        MvcResult r = mockMvc.perform(rb)
+            .andReturn(); // this could throw an exception
+        String tr = r.getResponse()
+            .getContentAsString();
+
+        ObjectMapper mapper = new ObjectMapper();
+        String er = mapper.writeValueAsString(userList.get(1));
+
+        assertEquals(er,
+            tr,
+            "Expected: " + er + "\nActual: " + tr);
+    }
+
+    @Test
+    void getUserLikeName() throws
+                           Exception {
+        String apiUrl = "/users/user/name/like/jam";
+
+        Mockito.when(userService.findByNameContaining(any(String.class)))
+            .thenReturn(userList);
+
+        RequestBuilder rb = MockMvcRequestBuilders.get(apiUrl)
+            .accept(MediaType.APPLICATION_JSON);
+        MvcResult r = mockMvc.perform(rb)
+            .andReturn(); // this could throw an exception
+        String tr = r.getResponse()
+            .getContentAsString();
+
+        ObjectMapper mapper = new ObjectMapper();
+        String er = mapper.writeValueAsString(userList);
+
+        assertEquals(er,
+            tr,
+            "Expected: " + er + "\nActual: " + tr);
+    }
+
+    @Test
+    void addNewUser() throws
+                      Exception {
+        String apiUrl = "/users/user";
+
+        Mockito.when(userService.save(any(User.class)))
+            .thenReturn(userList.get(0));
+
+        RequestBuilder rb = MockMvcRequestBuilders.post(apiUrl)
+            .contentType(MediaType.APPLICATION_JSON)
+            .accept(MediaType.APPLICATION_JSON)
+            .content("{\"username\": \"Priscilla\"}");
+
+        mockMvc.perform(rb)
+            .andExpect(status().isCreated())
+            .andDo(MockMvcResultHandlers.print());
+    }
+
+    @Test
+    void updateFullUser() throws
+                          Exception {
+        String apiUrl = "/users/user/{id}";
+
+        Mockito.when(userService.update(any(User.class),
+            any(Long.class)))
+            .thenReturn(userList.get(0));
+
+        //        {"userid":2,"username":"james","favcities":[]}
```

Comment:
Is this needed?

Reply:
I was using it for reference, I removed it when I did my revision earlier.